baseplate/context/thrift.py

```diff
@@ -52,25 +65,48 @@ def _enumerate_service_methods(client):
 class PooledClientProxy(object):
     """A proxy which acts like a thrift client but uses a connection pool."""
 
-    def __init__(self, client_cls, pool, root_span, namespace):
+    def __init__(self, client_cls, pool, root_span, namespace, retry_policy=None):
```

Comment:
`retry_policy=None` leading to `self.retry_policy=RetryPolicy.new(attempts=1)` seems potentially surprising. Maybe beef up the class docstring?